docs/nf4_science/rnaseq/03_multi-sample.md

```diff
@@ -0,0 +1,489 @@
+
+Let's create a module file called `modules/multiqc.nf` to house the `MULTIQC` process:
+
+```bash
+touch modules/multiqc.nf
+```
+
+Open the file in the code editor and copy the following code into it:
+
+```groovy title="modules/multiqc.nf" linenums="1"
+#!/usr/bin/env nextflow
+
+process MULTIQC {
+
+    container "community.wave.seqera.io/library/pip_multiqc:ad8f247edb55897c"
+    publishDir "results/multiqc", mode: 'copy'
+
+    input:
+    path '*'
+    val output_name
+
+    output:
+    path "${output_name}.html", emit: report
+    path "${output_name}_data", emit: data
+
+    script:
+    """
+    multiqc . -n ${output_name}.html
+    """
+}
+```
+
+### 2.2. Import the module into the workflow file
+
+Add the statement `include { MULTIQC } from './modules/multiqc.nf'` to the `rnaseq.nf` file:
+
+```groovy title="rnaseq.nf" linenums="3"
+// Module INCLUDE statements
+include { FASTQC } from './modules/fastqc.nf'
+include { TRIM_GALORE } from './modules/trim_galore.nf'
+include { HISAT2_ALIGN } from './modules/hisat2_align.nf'
+include { MULTIQC } from './modules/multiqc.nf'
+```
+
+### 2.3. Add a `report_id` parameter and give it a sensible default
+
+```groovy title="rnaseq.nf" linenums="9"
+/*
+ * Pipeline parameters
+ */
+params.hisat2_index_zip = "data/genome_index.tar.gz"
+params.report_id = "all_single-end"
+```
+
+### 2.4. Call the process on the outputs of the previous steps
+
+We need to give the `MULTIQC` process all the QC-related outputs from previous steps.
+
+For that, we're going to use the `.mix()` operator, which aggregates multiple channels into a single one.
+
+If we had four processes called A, B, C and D with a simple `.out` channel each, the syntax would look like this: `A.out.mix( B.out, C.out, D.out )`. As you can see, you apply it to the first of the channels you want to combine (doesn't matter which) and just add all the others, separated by commas, in the parenthesis that follows.
+
+In the case of our workflow, we have the following outputs to aggregate:
+
+- `FASTQC.out.zip`
+- `FASTQC.out.html`
+- `TRIM_GALORE.out.trimming_reports`
+- `TRIM_GALORE.out.fastqc_reports`
+- `HISAT2_ALIGN.out.log`
+
+So the syntax example becomes:
+
+```groovy title="Applying .mix() in the MULTIQC call"
+        FASTQC.out.zip.mix(
+        FASTQC.out.html,
+        TRIM_GALORE.out.trimming_reports,
+        TRIM_GALORE.out.fastqc_reports,
+        HISAT2_ALIGN.out.log
+        )
+```
+
+To this, we need to append the `collect()` operator in order to pull the reports for all the samples into a single call to `MULTIQC`, and also give it the `report_id` parameter:
+
+```groovy title="The completed MULTIQC call" linenums="33"
+    // Comprehensive QC report generation
+    MULTIQC(
+        FASTQC.out.zip.mix(
+        FASTQC.out.html,
+        TRIM_GALORE.out.trimming_reports,
+        TRIM_GALORE.out.fastqc_reports,
+        HISAT2_ALIGN.out.log
+        ).collect(),
+        params.report_id
+    )
```

Comment:
With beginners I might suggest splitting these calls for better clarity.
```suggestion
    // Comprehensive QC report generation
    multiqc_files_ch = Channel.empty.mix(
        FASTQC.out.zip,
        FASTQC.out.html,
        TRIM_GALORE.out.trimming_reports,
        TRIM_GALORE.out.fastqc_reports,
        HISAT2_ALIGN.out.log
        )
    multiqc_files_list = multiqc_files_ch.collect(),
    MULTIQC(
        multiqc_files_list,
        params.report_id
    )
```

Reply:
Ohh that is clearer, yes, thanks

Reply:
I'm going to keep this as an improvement for the next round because if I make this change there's a bunch of other files that need to be updated, and I don't quite have the runway for that right now. But it's a really good idea for explaining this bit more clearly.